bluebutton-data-pipeline-rif-load/src/test/java/gov/hhs/cms/bluebutton/data/pipeline/rif/load/RifLoaderIT.java

```diff
@@ -239,11 +248,41 @@ private static void assertAreInDatabase(LoadAppOptions options, EntityManagerFac
 					Object recordId = entityManagerFactory.getPersistenceUnitUtil().getIdentifier(record);
 					Object recordFromDb = entityManager.find(record.getClass(), recordId);
 					Assert.assertNotNull(recordFromDb);
+					assertAreInRifDataloadHistoryTable(entityManager, record, dataSetManifest);
 				}
 			}
 		} finally {
 			if (entityManager != null)
 				entityManager.close();
 		}
 	}
+
+	/**
+	 * Verifies that the specified RifDataloadHistory record is inserted actually
+	 * into the database based on what is in the the dataSetManifest
+	 * 
+	 * @param entityManager
+	 *            the {@link EntityManager} to use
+	 * @param record
+	 *            the RIF record to verify
+	 * @param dataSetManifest
+	 *            use the {@link DataSetManifest} to verify records are loaded into
+	 *            RifDataloadHistory table
+	 */
+	private static void assertAreInRifDataloadHistoryTable(EntityManager entityManager, Object record,
+			DataSetManifest dataSetManifest) {
+		if (record instanceof Beneficiary) {
+			RifDataloadHistory.PrimaryKey id = new RifDataloadHistory.PrimaryKey(RifFileType.BENEFICIARY.toString(),
+					Timestamp.from(dataSetManifest.getTimestamp()), dataSetManifest.getSequenceId());
+
+			Assert.assertNotNull(entityManager.find(RifDataloadHistory.class, id));
+		}
+		if (record instanceof CarrierClaim) {
+			RifDataloadHistory.PrimaryKey id = new RifDataloadHistory.PrimaryKey(RifFileType.CARRIER.toString(),
+					Timestamp.from(dataSetManifest.getTimestamp()), dataSetManifest.getSequenceId());
+
+			Assert.assertNotNull(entityManager.find(RifDataloadHistory.class, id));
+		}
```

Comment:
Can you make this an `if { ... } else if { ... } else { throw new BadCodeMonkeyException(); }` kind of deal, instead? Should help guard against false negatives in this test case, where things would/might blow up, except you hadn't tested them.

If you don't think the default "I don't handle this record type" case should blow up, you should definitely add a comment explaining why, so that the code doesn't look like a bug to future readers.

Reply:
Done